src/HttpClient.php

```diff
@@ -289,14 +60,23 @@ public function __construct($disableTls = false, $cafile = null)
      */
     public function download($url)
     {
-        $context = $this->getStreamContext($url);
-        $result = file_get_contents($url, null, $context);
+        if (extension_loaded('zlib')) {
+            array_push($this->headers, "Accept-Encoding: gzip\r\n");
```

Comment:
Shouldn't this be done in the constructor? Right now, for every call to `download()` the header will be added again.